truckersmp-cli

```diff
@@ -1,33 +1,75 @@
 #!/bin/sh
 
-if [ -z "$1" ]; then
-    echo "Usage: truckersmp-cli GAMEDIR"
-    exit
-fi
+readonly XDG_CACHE_HOME=${XDG_CACHE_HOME:-$HOME/.cache}
+
+usage() {
+    echo "Usage: ${0} [-d <path> -h] [GAMEDIR]"
+    echo "  -d path     mod directory, defaults to \$XDG_CACHE_HOME/truckersmp-cli"
+    echo "  -h          this help"
+    echo "  GAMEDIR     path to ETS2 od ATS game data, (optional: update only)"
+}
+
+quit() {
+    # clean up tmpfiles
+    rm "$filelist" "$checksums"
+    exit "$1"
+}
 
 listurl=https://update.ets2mp.com
 dlurl=https://downloads.ets2mp.com/files
 dlurlalt=https://download.ets2mp.com/files
 base="$(dirname "$0")"
-dldir="truckersmp"
+dldir="$XDG_CACHE_HOME/truckersmp-cli"
+filelist=$(mktemp)
+checksums=$(mktemp)
+updonly=false
+
+# fallback to old behavior
+if [ -d "$base/truckersmp" ]; then
+    dldir="$base/truckersmp"
+fi
+
+# https://stackoverflow.com/a/14203146
+OPTIND=1         # Reset in case getopts has been used previously in the shell.
+
+while getopts "d:h?" opt; do
+    case "$opt" in
+    d)	dldir=${OPTARG}
+        ;;
+    h|\?)
+        usage
+        quit 0
+        ;;
+    esac
+done
+
+shift $((OPTIND-1))
+[ "${1:-}" = "--" ] && shift
+
+if [ -z "$1" ]; then
```

Comment:
I feel like the default behaviour when called with no argument should be to display the usage message. Maybe you could add an option like -u for upgrade only ?